From b26cfbfabf97c6526aee11dfd21e2e20400756df Mon Sep 17 00:00:00 2001
From: David Edmundson <davidedmundson@kde.org>
Date: Wed, 29 Jul 2020 18:53:56 +0100
Subject: [PATCH] Client: Gracefully exit when the compositor closes

Currently if the compositor closes whilst a client is active the client
will qFatal leading to an assert and coredump handlers being run.

Comparatively on X we just call exit(). This brings wayland in line with
the XCB implementation.

Task-number: QTBUG-85631
Pick-to: 5.15
Change-Id: I316932291b314f3e7abab43c50e04abc2b1e87d7
---
 src/client/qwaylanddisplay.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylanddisplay.cpp b/src/client/qwaylanddisplay.cpp
index 9e535935..b60701ab 100644
--- a/src/client/qwaylanddisplay.cpp
+++ b/src/client/qwaylanddisplay.cpp
@@ -208,7 +208,8 @@ void QWaylandDisplay::checkError() const
     int ecode = wl_display_get_error(mDisplay);
     if ((ecode == EPIPE || ecode == ECONNRESET)) {
         // special case this to provide a nicer error
-        qFatal("The Wayland connection broke. Did the Wayland compositor die?");
+        qCritical("The Wayland connection broke. Did the Wayland compositor die?");
+        exit(1);
     } else {
         qFatal("The Wayland connection experienced a fatal error: %s", strerror(ecode));
     }
-- 
2.30.1

